### DISCUSSION TITLE
Error querying the database: db error: FATAL: remaining connection slots are reserved for non-replication superuser connections

### DISCUSSION THREAD

**Discussion by jaredcosulich** [supabase/supabase#5212] [Questions]
We have an app hosted on Vercel with Supabase providing the database. The app is in development and I am often the only one using it but I frequently get this error:

Error querying the database: db error: FATAL: remaining connection slots are reserved for non-replication superuser connections

When I do the app is completely unusable and it's not clear how to fix the problem.

Does anyone know what is going on here?

**Comment by jaredcosulich** [comment]
When I explore the Supabase documentation and search for "restart", an entry for restarting the database is offered but links to this page:

https://supabase.com/docs/blog/2020/08/02/continuous-postgresql-backup-walg#3-restart-the-database

Which is a 404.

**Reply by GaryAustin1** [reply]
Under UI Settings/General there is a restart server.  Not sure about your real issue.

**Reply by jaredcosulich** [reply]
Thank you that was help for restarting the server.

**Comment by jaredcosulich** [comment]
For future reference we needed to switch to the connection pooling connection string for connecting directly to the database. This was found under Database -> Connection Pooling (at the bottom of the page)

**Reply by salahhaciakilAc** [reply]
were you using the supabase client package ? if so, how did you switch ? is it by only changing the DATABASE_URL ? 
```
  return createBrowserClient<Database>(
    process.env.NEXT_PUBLIC_SUPABASE_URL!,
    process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!
  );
```

**Comment by bradgreens** [comment]
Thanks @jaredcosulich, maybe I missed this in the docs, but it could be more clear the Connection Pooling address is the way to go for active/ongoing development.
